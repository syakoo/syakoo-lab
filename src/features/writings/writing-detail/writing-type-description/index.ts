export { WritingTypeDescription } from "./writing-type-description";
