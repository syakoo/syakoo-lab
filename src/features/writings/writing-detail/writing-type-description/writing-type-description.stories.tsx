import type { Meta, StoryObj } from "@storybook/nextjs";

import { WritingTypeDescription } from ".";

const meta = {
  component: WritingTypeDescription,
  parameters: {
    layout: "fullscreen",
    testLevel: "snapshot",
  },
} satisfies Meta<typeof WritingTypeDescription>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {
  args: {
    type: "article",
  },
};
