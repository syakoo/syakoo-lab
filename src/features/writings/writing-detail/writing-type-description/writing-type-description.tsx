import type { WritingType } from "@/features/writings/_models/types";
import { writingTypeConfig } from "@/features/writings/_models/writing-type";
import { Icon } from "@/shared/design-system/icons";
import { Col, Row } from "@/shared/design-system/layout";
import { P, Text } from "@/shared/design-system/ui";

type WritingTypeDescriptionProps = {
  type: WritingType;
};

export const WritingTypeDescription: React.FC<WritingTypeDescriptionProps> = ({
  type,
}) => {
  const { iconName, label, description } = writingTypeConfig.find(
    (config) => config.type === type,
  )!;

  return (
    <Col as="section" gap="50">
      <Row align="center" gap="50">
        <Icon name={iconName} width="1em" />
        <Text weight="bold">{label}</Text>
      </Row>
      <div>
        <P color="secondary" size="75">
          {description}
        </P>
      </div>
    </Col>
  );
};
