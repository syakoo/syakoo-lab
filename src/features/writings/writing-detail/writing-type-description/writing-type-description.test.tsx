import { testStories } from "@/test/test-stories";

import * as stories from "./writing-type-description.stories";

describe("WritingTypeDescription", () => {
  testStories(stories);
});
