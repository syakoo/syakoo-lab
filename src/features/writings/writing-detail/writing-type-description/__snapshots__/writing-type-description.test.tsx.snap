// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritingTypeDescription スナップショット Sample 1`] = `
<body>
  <div>
    <section
      class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_column__zqca9tk flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
    >
      <div
        class="flex_flexStyle__zqca9ti flex_flexStyle_align_center__zqca9ts flex_flexStyle_variants_align_center__zqca9t7 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 14.25v-2.625a3.375 3.375 0 00-3.375-3.375h-1.5A1.125 1.125 0 0113.5 7.125v-1.5a3.375 3.375 0 00-3.375-3.375H8.25m0 12.75h7.5m-7.5 3H12M10.5 2.25H5.625c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125V11.25a9 9 0 00-9-9z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <p
          class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_100__1djq151l text_textStyle_variants_size_100__1djq1516 text_textStyle_weight_bold__1djq151t"
        >
          Article
        </p>
      </div>
      <div>
        <p
          class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_75__1djq151k text_textStyle_variants_size_75__1djq1515 text_textStyle_weight_normal__1djq151u"
        >
          Article は気合いの入った文書を残しています。
        </p>
      </div>
    </section>
  </div>
</body>
`;
